### DISCUSSION TITLE
UI improvements

### DISCUSSION THREAD

**Discussion by Falconbird18** [FreeTubeApp/FreeTube#5570] [General]
I feel like the current Freetube UI could use more work to polish it. Here is what I have been working on. Please give feedback and comments! 
![image](https://github.com/user-attachments/assets/c4a65803-4583-4637-a201-722aa5435812)
![image](https://github.com/user-attachments/assets/83bc6efb-6b90-4abf-9bbc-f7700276dbf3)

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Hi @Falconbird18 could you lay out the changes you made? The only ones that I see is making everything more rounded.

**Reply by Huecuva** [reply]
I don't know about graphical UI changges, but I think FreeTube could use a lot of QoL improvements. Why can't I just click on the three dot menu and add a video to Watch Later like I can in YouTube itself? I have to open a popup, potentially on top of what I'm watching, click on Watch Later and then save it. That's an extra click and annoying popup. Also, why can't I have my mouse cursor over the video itself with a massive white circle in the middle of the screen?

**Reply by absidue** [reply]
@Huecuva In the future please stay on topic instead of leaving a completely unrelated comment on a thread.

If you haven't already you can set a quick bookmark playlist on the playlists page, then it is just one click to add it to your quick bookmark playlist from the video thumbnail and watch page.

As for the play button, as mentioned in many other issues and threads, you can turn off the `Display Play Button in Video Player` setting in the Player settings if you don't want to see it.

**Reply by Huecuva** [reply]
Sorry about going off topic. I thought this might be a relevant thread about improvements to the app.

Thanks for the tips. Maybe I should have looked around more, but I figured those features would be so obvious they should be default. I guess I was mistaken.

If I could make another suggestions, might I suggest the addition of a feature similar to how YouTube will continue to play the video in a smaller box when you leave the page or scroll down? It's a pretty minor thing, but it is convenient to have a video playing while I'm looking through a channel or searching for other videos that I can then add to the playlist without interrupting it. The other two issues cleared up, that's probably my last remaining gripe with the app. Otherwise it's great. I only recently started using it.

**Comment by Falconbird18** [comment]
1. I made all corners have a 15px border radius.
2.  I removed almost all box shadows. 
3. I made the background color the same.
4. I added more variables to themes.css
5. I worked on the side bar, the scroll bar only shows when you hover over the side bar (This is still not perfect yet).
These are all the main changes I have done, I could find more if you are interested.
Old version:
![image](https://github.com/user-attachments/assets/ffc374d1-aac5-415b-b0c9-494006cad197)
New version:
![image](https://github.com/user-attachments/assets/cd02643a-dd99-431c-a9d9-469ffebed8fd)

**Reply by absidue** [reply]
Maybe some of the suggestions could be considered but as with the massive design overhauls proposed in the past by other people, they seem to sacrifice usability and accessibility for the sake of making it look more trendy.

E.g. removing all visual differences between the top bar, side bar and content makes the app much harder to use, if you want them to all be the same color, you need to add borders or shadows between them so that they are still clearly visually separated.

I also don't like the sound of even more theming variables, it's already hard enough to add a new theme as it is (the theme pull requests tend to go through lots of review cycles and then are usually abandoned by their creators), so please try to avoid making the theming even more complicated.

**Reply by Falconbird18** [reply]
Thank you for your great feedback!
I absolutely agree that the user experience should be the first priority. You are completely right that the sidebar and the top bar should have some separation. I feel a bit stupid to not have realized that.

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
I appreciate the list and thankyou for understanding our points about the UX. Could you maybe explain in more detail why you made the changes that you did.

**Reply by failafail8** [reply]
i do really like the rounded thumbnails. 
Not so much to lack of contrast for the side and top bar.
looks pretty, well done.

**Comment by Falconbird18** [comment]
Sorry for the late reply,
One of the reasons I think that the FreeTube UI needs some redesign is because of all the shadows. 
Before
![image](https://github.com/user-attachments/assets/55d48064-4cf5-4ed9-9989-7c4b17eb388a)
For example, the refresh button for the subscriptions has shadows all along the bar. I think the shadows can be distracting, and can separate
the whole bar from the background. 
After
![image](https://github.com/user-attachments/assets/e3af816b-adfa-401d-a26b-1b1ebe222c10)
Another change I made was to remove the cards (I just changed the background color), to unify the background color.
Before
![image](https://github.com/user-attachments/assets/d8715ad1-d01d-4334-b612-6620acdcd1dd)
Notice how the card around a video changes color when you hover on it. I thought this was necessary to really show
the video was selected and would play when clicked on. Before, only the cursor changed.
After.
![image](https://github.com/user-attachments/assets/3a5f06a3-99de-4f7d-a209-83fcadf602b3)
I also changed the header and sidebar. The design was ok before, but in my personal opinion, my version looks better. 
Before
![image](https://github.com/user-attachments/assets/56a13eda-b02a-497c-b2c9-45fac18c1643)
After
![image](https://github.com/user-attachments/assets/3bef6f86-94d9-4526-ad7c-fc892fdd5d49)
Please note that these are my personal opinions, and any feedback or suggestions are welcome.
Thanks everyone

**Reply by absidue** [reply]
> For example, the refresh button for the subscriptions has shadows all along the bar. I think the shadows can be distracting, and can separate the whole bar from the background.

That's intentional it's not meant to blend into the background, but from looking at the rest of the screenshots blending everything together into one big thing even though they are completely separate UI elements with different purposes, seems to be your goal.

**Comment by Falconbird18** [comment]
I personally disagree with you, but I will be happy to improve the UI in a way you like. If that means discarding my changes and reverting back to the old UI and going from there, that's ok with me.

**Comment by absidue** [comment]
Things that slightly concern me about this:
* It looks like you are going for cloning YouTube's UI, which isn't inherently a bad thing, but it does mean that things that don't have an equivalent in YouTube's UI, like the last refresh timestamp, are just weirdly placed above the title, instead of being properly integrated into the UI.
* In your screenshots you seem to have got ridden of any indication that a video has been watched, I'll give you the benefit of doubt here and assume you just forgot to take a screenshot that included them, as I hope you didn't think that removing features in the name of design was a good idea.

**Comment by Falconbird18** [comment]
Here is a screenshot with a watched video. 
![image](https://github.com/user-attachments/assets/dc0331dc-5a21-4fae-975c-7336ee4e2db9)
I wasn't trying to clone the YouTube UI, and I think FreeTube should be different, but easy to get used to. 
Do you have particular preferences for UI? What exactly is your idea or vision for the UI? Knowing this will help me improve the UI and UX in 
a way you are happy with.
Thanks

**Reply by kommunarr** [reply]
Hi @Falconbird18, taking a look at this now.
- With regards to the rounding, this is something I've looked into before, but it's got some awkwardness to implement that would take some time & effort to address. To list a couple of examples I can see in your image: the junction point of the side nav and the top nav, as well as the video progress bar protruding with rounded bottom edges.
- As absidue mentioned, the video refresh section does blend into the background without any shadow or divider there. This one seems like a non starter.
- As abisdue also mentioned, moving the video refresh timestamp like this does not seem to be providing any benefits, and it's in fact a bit more jarring since it's still misaligned with the heading.
- The remaining unique aspect of making the background color the same is a change I like, although we're pretty much accomplishing this in an only slightly different way in #5294.

With all that said, though, we appreciate your enthusiasm and hope you can pursue more ways to support and improve the project.

**Comment by Falconbird18** [comment]
Yeah, I do agree with you that at least the rounding I have done is too much. 
And I am aware that the video refresh timestamp is misaligned. It's an issue that I haven't fixed yet, but is on my radar. 
Also, thank you so much for the encouraging response!

**Comment by lamborghinipth** [comment]
performance is the most important -> shouldn't rounding corners, using borders, using shadows or using too much CSS style.

**Comment by GCAT01** [comment]
I would like to chime in as someone with a visual disability. You are going for the clean, unbroken look here and I get it. Smoothing out UIs is all the rage right now and it DOES genuinely look nice. However, those little drop shadows and bits of separation are what help me distinguish parts of the screen. Having that slight color difference between the headers, main area, and scrollbar immediately tell me that those are all separate elements that are potentially intractable or, at least, serve different purposes. Flat UIs make it very difficult for me to tell where the header ends and the viewing area begins or if the header even exists. You may not think that's a big deal, but looking at a window and not being able to visually understand the layout is extremely frustrating.

The constantly visual scrollbars also help with this. I can immediately tell that **this part** of the screen is a separate element and scrollable. It also helps me line up the mouse with where I need to be to click/drag.

I get what you're going for but, for me, UIs like this can be really frustrating and diminish my user experience.

**Comment by Falconbird18** [comment]
@GCAT01 I’ve been thinking about the UI design (especially after taking a graphic design course), and I think the main issue I have with it is the lack of consistency. For instance, the top bar and the refresh button have shadows, but very few other elements do.  

To improve the UI, I think we should settle on a clear design style and apply it uniformly across the interface. Right now, I’m experimenting in Inkscape to explore some ideas.

**Reply by pkrasicki** [reply]
The shadows have different styles too. The more button on a video (three horizontal dots) brings up a dropdown menu that has a different shadow from the profile selection menu (when you click on your profile in top right corner).

Also tabs have different length depending on a page.

**Comment by pkrasicki** [comment]
Here is my design:

![freetube-restyled](https://github.com/user-attachments/assets/5f586196-4437-49cc-9aa0-16bd9bd5dac1)

![freetube-restyled2](https://github.com/user-attachments/assets/f0cd274d-063b-4f58-a599-039accb13e4e)

You don't need shadows or backgrounds or borders to be able to tell things apart. For example, you can clearly tell that each video on the list is a separate thing despite there not being any clear boundary between them. You are able to distinguish them simply because of spacing. Same with other stuff - group things that are related and space them from things that aren't. No backgrounds needed. And you don't need to know where exactly the top bar ends. Why would you need that? You just need to be able to distinguish buttons and the search bar - the things you are going to interact with.

We don't do this just because it looks nice. It's also more efficient (easier to read) once you get used to it. That's why most modern apps looks like this. Try it for 10 days (I can post a git patch for Freetube 0.21.3), satisfaction guaranteed 😀.

**Reply by Falconbird18** [reply]
I think your version looks way better, but I still have some issues with it (and the official app too).
- First, the accent color alternates between red and blue. One page might have a red accent color, while another page might have blue.
- Second, the mix of styles is a bit odd. Your version fixes most of the problems, but the different amount of rounding is strange. Buttons and the search bar are rounded, but nothing else.

Additionally, I would like to respectively point out that borders and shadows are commonly used (just look at GitHub), and different background colors for various elements are normal (see any Google or Microsoft app). Despite this, I'm fine with having one background color for FreeTube as I think it works well.
Overall though your design is a huge improvement!

**Reply by pkrasicki** [reply]
Thanks!

>First, the accent color alternates between red and blue. One page might have a red accent color, while another page might have blue.

They use red as primary color and blue as secondary color. Primary color shouldn't be used this much, but in FreeTube it's used a lot. YouTube used to do the same [around 2017](https://www.webdesignmuseum.org/gallery/youtube-in-2017), which is probably what FreeTube's design is mostly based on. In my design I replaced those colors with white (black in light theme) and gray. I think YouTube has similar colors nowadays.

![freetube-restyled3](https://github.com/user-attachments/assets/e9e35ca1-2ff1-4f6a-85d4-de143f022649)

>Second, the mix of styles is a bit odd. Your version fixes most of the problems, but the different amount of rounding is strange. Buttons and the search bar are rounded, but nothing else.

I think you're saying that the thumbnails should be rounded? I don't think it's necessary, but I'm not against it. YouTube, Odysee, nebula.tv and [PeerTube](https://diode.zone) have rounded thumbnails. But piped.video, pornhub and xvideos (I ran out of ideas for popular sites to check) don't have rounded thumbnails.

>Additionally, I would like to respectively point out that borders and shadows are commonly used (just look at GitHub), and different background colors for various elements are normal (see any Google or Microsoft app).

YouTube, piped, nebula and other video sites I mentioned don't really do that. Odysee and PeerTube have backgrounds in a few places. I think that GitHub overuses borders - try comparing it to Reddit for example. In my design I think I only kept shadows on the floating refresh button, dropdowns and popups. I'm not saying that shadows are bad, just that they are currently overused in FreeTube.

**Reply by pkrasicki** [reply]
Maybe we will get PR #5294 merged eventually and that would be a step in the right direction.

Here are my other proposals if you're interested:
- https://github.com/FreeTubeApp/FreeTube/issues/5303
- https://github.com/FreeTubeApp/FreeTube/discussions/6294
- https://github.com/FreeTubeApp/FreeTube/discussions/6293

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
> You don't need shadows or backgrounds or borders to be able to tell things apart. For example, you can clearly tell that each video on the list is a separate thing despite there not being any clear boundary between them. You are able to distinguish them simply because of spacing. Same with other stuff - group things that are related and space them from things that aren't. No backgrounds needed. And you don't need to know where exactly the top bar ends. Why would you need that? You just need to be able to distinguish buttons and the search bar - the things you are going to interact with.

> We don't do this just because it looks nice. It's also more efficient (easier to read) once you get used to it. That's why most modern apps looks like this. Try it for 10 days (I can post a git patch for Freetube 0.21.3), satisfaction guaranteed 😀.

@pkrasicki maybe you should read https://github.com/FreeTubeApp/FreeTube/discussions/5570#discussioncomment-11417373

**Reply by pkrasicki** [reply]
@efb4f5ff-1298-471a-8973-3d47447115dc That's what I was partially responding to. Do you believe that all popular video sites have terrible accessibility? Because you will have a hard time finding one that's popular and looks like FreeTube. And if you do believe that or think that modern user interfaces are bad, the conclusion would have to be that most UI designers are incompetent. And that would be silly. Also, my version of dark theme has better contrast than current dark theme (@Falconbird18's version does too), which is a good thing for accessibility.

I suspect the real issue here is that those changes are too big - it would be annoying for users if we change so much at once. Perhaps we should make those changes gradually over a few releases to make it easier for users to adjust to them.

**Reply by kommunarr** [reply]
I'll bring in @GCAT01 to the discussion if they want to try out the design patch you have and assess their experience. I do want to call out that the popularity of a design trend does imply that it's being done for a considerable reason, but it doesn't necessarily mean that it's 10/10 on every category like digital accessibility. If the goal is to keep emulating what the modern YT app looks like, we will have to keep implementing broad redesigns whenever they implement new broad redesigns of their UI. In reality, our goals should be to deliver the best user experience we can to people, and that involves listening to our dis/abled users and being very intentional & evidence-backed with the changes we do decide to implement.

**Reply by pkrasicki** [reply]
The goal should be to emulate what modern video applications look like (you don't have to limit yourselves to yt and I assume you don't want to copy them exactly, since you probably don't want your project to look like a clone, instead you want it to be unique) and that will deliver the best user experience to people. There are [accessibility standards](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG) and if you want to start paying more attention to that, that's great, you can do that (whichever level of compliance you want). I'm not an accessibility expert, so if you notice that my design doesn't conform to some of the criteria you are interested in, you can say it and we will try to fix it.

>that involves listening to our dis/abled users and being very intentional & evidence-backed with the changes we do decide to implement.

Ideally you would reach out to UI experts from other projects (since you are so skeptical of anything new), listen to what they have to say, maybe get someone to make you a design (if they have better proposals) and follow that as closely as you can and keep doing that when adding new features. Then listen to the feedback from the users (keeping in mind that most users are not experts and that they might be biased).

**Reply by pkrasicki** [reply]
Here is my Restyled branch: https://github.com/pkrasicki/FreeTube/tree/v0.21.3-restyled. This is currently the latest version, it's based on FreeTube 0.21.3.

**Reply by pkrasicki** [reply]
@efb4f5ff-1298-471a-8973-3d47447115dc @kommunarr I have a solution for you. Add a settings section like this (without changing our current design for settings):

![gnome-accessibility](https://github.com/user-attachments/assets/fc7a0876-6455-411f-8f2f-4450b8df67ca)

![gnome-accessibility2](https://github.com/user-attachments/assets/3bb90a47-0c8c-452f-b081-457f389cc587)

**Comment by mjones129** [comment]
Thanks to everyone who has put in the work to make the UI what it is. It's really awesome, I use FreeTube daily. Something I came across just today, (in Freetube v0.22.1 Beta running on Windows 11) was when I'm using the autoplay feature, and the current video ends, there's a little toast message that appears for 5 seconds before the next video plays (which I really like) but the toast itself fades in and out every second and it's actually kinda difficult to click on to cancel autoplay via this toast. Perhaps just making the toast solid and hold for 5 seconds instead of fading up and down would make it a bit easier to interact with. Screenshot of the toast message in question:
![image](https://github.com/user-attachments/assets/e4cb7890-df15-4936-88f4-035eae0be450)

**Reply by absidue** [reply]
@mjones129 That is already covered by #1214, please check for existing issues before leaving comments on unrelated threads.

**Comment by alihashimxd** [comment]
Sorry I am not a dev but why can't there be a setting to make the UI more rounded and an additional warning with it that it does impact the performance? All of the guys above me who are suggesting UI improvemnts have one thing in common, rounded corners. Personally just making the UI rounded makes it look way too cool and modern modern. Please add it as a setting option that would be really cool :))) 
here's my example which i made using my friend's help we basically edited the css file to round the corners:
![Screenshot (75)](https://github.com/user-attachments/assets/4d5ea471-d2c4-4c3b-b745-1cb3852cff79)
![Screenshot (74)](https://github.com/user-attachments/assets/b4bf62ca-a142-4724-bd56-ccfb629fd976)
![Screenshot (73)](https://github.com/user-attachments/assets/131a117c-23d6-409b-99be-e1b837963e6b)

**Reply by kommunarr** [reply]
Hi @AliTempXD, [see my earlier comment here](https://github.com/FreeTubeApp/FreeTube/discussions/5570#discussioncomment-10572162) on rounding:

> With regards to the rounding, this is something I've looked into before, but it's got some awkwardness to implement that would take some time & effort to address. To list a couple of examples I can see in your image: the junction point of the side nav and the top nav, as well as the video progress bar protruding with rounded bottom edges.

**Comment by failafail8** [comment]
its good to know that i have zero coding and zero designing and low english skills, but im a heavy freetube user so ill share my thoughts:

i've read every conversations about the play/pause button size and opacity for desktop and i think it should be disabled by default. 
I don't know if it is more a QOL or UI suggestion tho.

What strikes me first about the ui is this shadow around the update line.
In my opinion the profile selection  button should be aligned on the left menu using the same kind of icons and the update button should take its place on the top right and the line "Videos feed last updated:" should be replace just by the time

I also think the category should be centered and written in upper case to fit the rest.

No joke it took me almost 30 minutes to edit my screenshot on gimp to make my opinion a little bit more clear
 
![Copie d'écran_20250131_050803](https://github.com/user-attachments/assets/7e8cfd5d-b9d9-4caf-afbf-fe2bad8d0d0b)

![b](https://github.com/user-attachments/assets/9c3a4114-efb7-42d1-bb56-93d5b82c10ab)


Since i never post on github, if anyone from the dev community read me : Thank you for the work !
(edited again the second picture)

**Reply by pkrasicki** [reply]
I have a different [proposal](https://github.com/FreeTubeApp/FreeTube/discussions/6293#discussioncomment-11643324) for the update time section:

![refresh-section3-cut](https://github.com/user-attachments/assets/beea639e-6796-46e2-88f4-4f805c4cf17d)

But your design points out something important: the refresh button should be available in the navbar and usable on every page.

**Comment by failafail8** [comment]
After the update i rechecked my settings, and i had an idea that i guess is doable without too much difficulties or bad effects (i may be wrong).
When u select the primary and secondary color the hovering could be made of the concerned color. Just to give some ease of use.
![1](https://github.com/user-attachments/assets/aabe9aea-5855-4683-8e88-2295793643cd)

**Comment by VictorVow** [comment]
My main UI issue is with the Profiles dropdown. I wish it had nesting like PocketTube: 

![image](https://github.com/user-attachments/assets/4f97f089-dc0d-43da-9679-0d8d2e9d9702)

**Comment by vot** [comment]
Hi all, the other day I started looking at making some very similar improvements myself and started poking away at a few things that bothered me. I think we could do with a little wider set of improvements, i.e. I was more concentrating on the user journeys and general usability than the design.

**Screenshots:**
<br />
Screenshot 1: Subscriptions page
<img width="3070" height="2108" alt="Screenshot 2026-09-25 at 18 28 52" src="https://github.com/user-attachments/assets/3df24039-addc-4880-8a35-4c7eb176cc60" />
<br />
Screenshot 2: Updated Settings page 
<img width="3070" height="2108" alt="Screenshot 2026-09-25 at 18 09 10" src="https://github.com/user-attachments/assets/49c04c44-d34c-45e9-87e3-aa6ae3a5acf9" />
<br />
Screenshot 3: Updated Profile Manager
<img width="3070" height="2108" alt="Screenshot 2026-09-25 at 18 11 44" src="https://github.com/user-attachments/assets/b3f02f69-c2e4-4968-b6c6-3e3360b7d1dc" />
<br />
Screenshot 4: Profile Manager - copy subscription modal
<img width="3070" height="2108" alt="Screenshot 2026-09-25 at 18 13 05" src="https://github.com/user-attachments/assets/dfe26703-4c04-4b57-a147-fffcd0b6f379" />
<br />
**List of changes:**
- rounded corners (obviously :D )
- floating nav bar on the left - it remains visually distinctive as a separate element but looks a bit fresher. It also has the expand/contract button integrated into it instead of the "hamburger" icon in the top nav
- the "feed last updated" line is now replaced with a refresh button in the top nav / toolbar
- Settings and about pages are now a single page with tabs to make the settings easier to navigate (also settings button moved to the top nav)
- Profile Manager page completely redone to make it a bit more user friendly. The "Other channels" section completely gone, replaced by "Copy to" functionality


I hope to get this merged in at some point so looking for feedback. (I'd also like to go further in an overall UI refresh but that's a much larger discussion - i.e. looking into adding tabs, tackling some of the issues raised by other people, etc).

**Reply by vot** [reply]
@efb4f5ff-1298-471a-8973-3d47447115dc and @absidue - tagging you guys directly as I'm looking for feedback.

I'm hoping to do a fair bit of UI/UX work throughout the app. I'd like to update a lot of user journeys, info organisation, user experience, etc but those are some potentially large changes so I'd like to make sure that it's going in the right direction before taking this further. Between some of my own needs and issues currently raised that could take a while and make sizeable changes to the app.

The code I've got on the branch needs cleanup of course but just wondering how far I can take this and still have a possibility of getting it merged :)
